include/libstatistics_collector/topic_statistics_collector/received_message_age.hpp

```diff
@@ -41,11 +41,12 @@ template<typename M, typename = void>
 struct HasHeader : public std::false_type {};
 
 /**
- * True if the message has a header
+ * True if the message has a header of type Header
  * @tparam M
  */
 template<typename M>
-struct HasHeader<M, decltype((void) M::header)>: std::true_type {};
+struct HasHeader<M, typename std::enable_if<std::is_same<std_msgs::msg::Header,
```

Comment:
You will definitely need to `#include std_msgs/msg/header.hpp` in this file - any given file that is including this header will not necessarily have included it. "Include what you use".

You will probably also now need to include `std_msgs` as an ament dependency of libstatistics_collector in the CMakeLists, so that other packages that consume this library have access to its headers. It was included as an `exec_depend` and `build_depend`, but is not currently mentioned in `ament_target_dependencies` for `libstatistics_collector`

Reply:
I tried that yesterday [here](https://github.com/ros-tooling/libstatistics_collector/tree/dabonnie/header-check) (just pushed), but something is still missing.

Reply:
I think the only problem in your attempt was that you included the C header `header.h` instead of the C++ header `header.hpp`, which declares the `std_msgs` namespace. I just pushed commit https://github.com/ros-tooling/libstatistics_collector/commit/83486f2dc4a29e9dadff1e9caea836cde63c192b to your branch and I was able to successfully `colcon build --packages-up-to rclcpp`